Scripts/Python/xIniInput.py

```diff
@@ -60,19 +62,48 @@
 kTurnLeft = '"Turn Left"'
 kTurnRight = '"Turn Right"'
 kJump = '"Jump Key"'
+kStrafeLeft = '"Strafe Left"'
+kStrafeRight = '"Strafe Right"'
 kExit = '"Exit Mode"'
-kExiGUI = '"Exit GUI Mode"'
+kToggleFP = '"Toggle First Person"'
+kOpenYB = '"Game.KIOpenYeeshaBook"'
+kHelp = '"Game.KIHelp"'
+kOpenKi = '"Game.KIOpenKI"'
+kTakePicture = '"Game.KITakePicture"'
+kKICreateJournal = '"Game.KICreateJournal"'
 kTalk = '"Push to talk"'
+kChat = '"Game.EnterChatMode"'
+kCreateMarkerFolder = '"Game.KICreateMarkerFolder"'
+kCreateMarker = '"Game.KICreateMarker"'
+
+kIniArray = [ kWalkForward, kWalkBack, kTurnLeft, kTurnRight, kJump, kStrafeLeft, kStrafeRight, kExit, kToggleFP, kOpenYB, kHelp, kOpenKi, kTakePicture, kKICreateJournal, kTalk, kChat, kCreateMarkerFolder, kCreateMarker ]
+
+
 
+def ConstructFilenameAndPath():
+    global gFilenameAndPath
+    if gFilenameAndPath == "":
+        if PtIsInternalRelease():
+            # check for local file
+            localNameAndPath = "init/" + gFilename
+            if PtFileExists(localNameAndPath):
+                gFilenameAndPath = localNameAndPath
+                PtDebugPrint("xIniInput::ConstructFilenameAndPath(): Using internal \"" + gFilenameAndPath + "\" file")
+                return
+        # otherwise, use the standard init path
+        gFilenameAndPath = PtGetInitPath() + "/" + gFilename
```

Comment:
This should probably return the path instead of stashing it as a global variable. That way, you can get rid of the `gFilenameAndPath` thing and just do `gIniFile.writeFile(ConstructFilenameAndPath())`.